packages/di/src/common/decorators/lazyInject.ts

```
@@ -10,42 +34,26 @@ import {injector} from "../fn/injector.js";
-  key: string,
-  resolver: () => any,
-  {optional = false, packageName = resolver.toString()}: {optional?: boolean; packageName?: string} = {}
-): PropertyDecorator {
-  return (target: any, propertyKey: string | symbol): any | void => {
-    let bean: any, token: any;
+export function LazyInject(resolver: () => Promise<{default: unknown}>): PropertyDecorator;
+export function LazyInject(key: string, resolver: () => Promise<{default: unknown}>): PropertyDecorator;
+export function LazyInject(...args: any[]): PropertyDecorator {
+  let resolver = mapOptions(args);
 
+  return (target: any, propertyKey: string | symbol): any | void => {
```

Biome

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)